---
title: "dogma"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{dogma}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(dogma.package)
```

## Summary

The dogma package is a collection of functions which enables the user to replicates the functions of the central dogma of molecular biology from generating the DNA sequence, transcription to translation. It can provide the count of all amino acids generated in a barplot. 

#### Data

The data used by the package includes a standard translation table for translating codons into amino acids.


#### Function 1: Create DNA sequence

The create_DNA() function returns a random DNA sequence of a length specify by the user. The new sequence will be formed by A, T, G and C nucleotides.
Input: n <-- is the number (integer) of characters for defining the length of the sequence.
Output: The string with length of n containing the nucleotides in a random order.
```{r}
Create_DNA <- function(n){
     nucleotides <- sample(c("A", "T", "G", "C"), size = n, replace = TRUE)
     DNA_sequence <- paste0(nucleotides, collapse = "")
     return(DNA_sequence)}
```
Example:
```{r}
create_DNA(4)
```


#### Function 2: Transcription from DNA to RNA

The dna_to_RNA() function converts a DNA sequence into an RNA sequence by replacing all T (thymine) with U (uracil), both uppercase and lowercase. It uses the gsub() function to make the substitutions and returns the transcribed RNA sequence.

**Argument and output**
Input: DNA sequence as a string
Output: RNA sequence as a string

**Function**
```{r}

dna_to_RNA <- function(dna_sequence){
  rna_sequence <- gsub("T", "U", dna_sequence)
  rna_sequence <- gsub("t", "u", rna_sequence)
  return(rna_sequence)
}

```


#### Function 3: Going from RNA to Codons

The function mRNA_to_codon takes RNA in a string as input to extract and replace with codons. The output of the function is a vector with multiple codons as strings seperated by ",".

**Example Usage**
```{r,message=FALSE}
mRNA <- "AUGUCG"
mRNA_to_codon(mRNA)
```


#### Function 4: 

The function takes a vector with codons (output from function 3), and using the codon table, generates a string of amino acids. 

#### Function 5: Count AA and present in barplot

The function takes a string of amino acids (one letter abbreviations), counts the number of each amino acid, and generates a barplot showing the counts. The output of the function is the barplot. The function depends on the packages ggplot2 and stringr.

The function is performed by typing "aa_plot(data)" with data being a string of amino acids.
